#!/bin/sh -e
## 02_ClntCfgAddr.dpatch by Tomasz Mrugalski
##
## DP: Fix for a possible problem when compiling with gcc 4.x

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

case "$1" in
       -patch) patch $patch_opts -p0 < $0;;
       -unpatch) patch $patch_opts -p0 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
--- ClntCfgMgr/ClntCfgAddr.cpp	2004-10-25 22:45:52.000000000 +0200
+++ ../dibbler-cvs/ClntCfgMgr/ClntCfgAddr.cpp	2005-11-03 22:23:43.000000000 +0100
@@ -54,7 +57,7 @@
 TClntCfgAddr::TClntCfgAddr() {
     this->Valid=ULONG_MAX;
     this->Pref=ULONG_MAX;
-    Addr=0;
+    Addr= new TIPv6Addr();
 }
 
 TClntCfgAddr::TClntCfgAddr(SmartPtr<TIPv6Addr> addr) {
